src/utils/report.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * First draft for error code categories and values.
+ * Very much up for discussion.
```

Comment:
Looks like a good start :) might be good to add a note about how this relates to setErrorMap when adding new codes. Also might be good to keep CODES and setErrorMap in the same place since it looks like you need to add a CODE and manually map it in setErrorMap